boot/seal.go

```diff
@@ -531,74 +518,17 @@ func resealKeyToModeenvForMethod(method device.SealingMethod, rootdir string, mo
 	return resealKeyForBootChains(method, rootdir, params, expectReseal)
 }
 
-func resealKeyForBootChainsImpl(method device.SealingMethod, rootdir string, params *resealKeyForBootChainsParams, expectReseal bool) error {
-	switch method {
-	case device.SealingMethodFDESetupHook:
-		// FIXME: do something
-		return nil
-	case device.SealingMethodTPM, device.SealingMethodLegacyTPM:
-	default:
-		return fmt.Errorf("unknown key sealing method: %q", method)
-	}
-
-	saveFDEDir := dirs.SnapFDEDirUnderSave(dirs.SnapSaveDirUnder(rootdir))
-	authKeyFile := filepath.Join(saveFDEDir, "tpm-policy-auth-key")
-
-	// reseal the run object
-	pbc := toPredictableBootChains(append(params.runModeBootChains, params.recoveryBootChainsForRunKey...))
-
-	needed, nextCount, err := isResealNeeded(pbc, bootChainsFileUnder(rootdir), expectReseal)
-	if err != nil {
-		return err
-	}
-	if needed {
-		pbcJSON, _ := json.Marshal(pbc)
-		logger.Debugf("resealing (%d) to boot chains: %s", nextCount, pbcJSON)
-
-		if err := resealRunObjectKeys(pbc, authKeyFile, params.roleToBlName); err != nil {
-			return err
-		}
-		logger.Debugf("resealing (%d) succeeded", nextCount)
-
-		bootChainsPath := bootChainsFileUnder(rootdir)
-		if err := writeBootChains(pbc, bootChainsPath, nextCount); err != nil {
-			return err
-		}
-	} else {
-		logger.Debugf("reseal not necessary")
-	}
-
-	// reseal the fallback object
-	rpbc := toPredictableBootChains(params.recoveryBootChains)
-
-	var nextFallbackCount int
-	needed, nextFallbackCount, err = isResealNeeded(rpbc, recoveryBootChainsFileUnder(rootdir), expectReseal)
-	if err != nil {
-		return err
-	}
-	if needed {
-		rpbcJSON, _ := json.Marshal(rpbc)
-		logger.Debugf("resealing (%d) to recovery boot chains: %s", nextFallbackCount, rpbcJSON)
-
-		if err := resealFallbackObjectKeys(rpbc, authKeyFile, params.roleToBlName); err != nil {
-			return err
-		}
-		logger.Debugf("fallback resealing (%d) succeeded", nextFallbackCount)
-
-		recoveryBootChainsPath := recoveryBootChainsFileUnder(rootdir)
-		if err := writeBootChains(rpbc, recoveryBootChainsPath, nextFallbackCount); err != nil {
-			return err
-		}
-	} else {
-		logger.Debugf("fallback reseal not necessary")
-	}
-
-	return nil
+func resealKeyForBootChainsImpl(method device.SealingMethod, rootdir string, params *ResealKeyForBootChainsParams, expectReseal bool) error {
+	return fmt.Errorf("FDE manager was not started")
 }
 
 var resealKeyForBootChains = resealKeyForBootChainsImpl
 
-func MockResealKeyForBootChains(f func(method device.SealingMethod, rootdir string, params *resealKeyForBootChainsParams, expectReseal bool) error) (restore func()) {
+func InstallResealKeyForBootChains(f func(method device.SealingMethod, rootdir string, params *ResealKeyForBootChainsParams, expectReseal bool) error) {
```

Comment:
we usually just have a top level variable with a func value without an install function fwiw